shelley/chain-and-ledger/formal-spec/update.tex

```diff
@@ -0,0 +1,396 @@
+
+
+The $\mathsf{UPDATE}$ transition is responsible for federated governance model in Shelly.
+This includes the ability to change the protocol parameters and also handles software
+updates.
+
+The application versions behave much like a protocol parameter, but because adoption
+for them must occur as soon as consensus is met, they are handled separately.
+Therefore the $\mathsf{UPDATE}$ is the combination of the
+Therefore the $\mathsf{PPUP}$ transition for the protocol parameters and the
+the $\mathsf{AVUP}$ transition for the application versions.
+
+Note that the signature for the keys in the proposal will be checked in the
+$\mathsf{UTXOW}$ transition.
+
+
+\begin{figure}[htb]
+  \emph{Protocol Parameter Update environment}
+  \begin{equation*}
+    \PPUpdateEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{slot} & \Slot & \text{current slot}\\
+        \var{dms} & \VKeyGen\mapsto\VKey & \text{genesis key delegations} \\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Protocol Parameter Update transitions}
+  \begin{equation*}
+    \_ \vdash
+    \var{\_} \trans{ppup}{\_} \var{\_}
+    \subseteq \powerset (\PPUpdateEnv \times \PPUpdate \times \PPUpdate \times \PPUpdate)
+  \end{equation*}
+  %
+  \caption{Protocol Parameter Update transition-system types}
+  \label{fig:ts-types:pp-update}
+\end{figure}
+
+\begin{figure}[htb]
+  \begin{equation}\label{eq:pp-update-Empty}
+    \inference[PP-Update-Empty]
+    {
+      \var{pup} = \emptyset
+    }
+    {
+      \begin{array}{l}
+        \var{slot}\\
+        \var{dms}\\
+      \end{array}
+      \vdash \var{pup_s}\trans{ppup}{pup}\var{pup_s}
+    }
+  \end{equation}
+
+  \nextdef
+
+  \begin{equation}\label{eq:update-nonempty}
+    \inference[PP-Update-Nonempty]
+    {
+      \var{pup}\neq\emptyset
+      &
+      \dom{pup}\subseteq\dom{dms}
+      \\
+      \var{slot} < \firstSlot{((\epoch{slot}) + 1) - \SlotsPrior}
+    }
+    {
+      \begin{array}{l}
+        \var{slot}\\
+        \var{dms}\\
+      \end{array}
+      \vdash
+      \var{pup_s}
+      \trans{ppup}{pup}
+      \varUpdate{pup_s\unionoverrideRight pup}
+    }
+  \end{equation}
+
+  \caption{Protocol Parameter Update inference rules}
+  \label{fig:rules:pp-update}
+\end{figure}
+
+\clearpage
+
+Figure~\ref{fig:funcs:helper-updates} gives some helper functions for the
+application version update transition.
+The function $\fun{votedValue_T}$ returns
+the consensus value of update proposals in the event that at least five
+genesis keys agree.
+This function will also be used for the protocol parameters later in Section~\ref{sec:epoch}.
+Note that $\type{T}$ is an arbitrary type.
+
+%%
+%% Figure - Helper Function for Consensus of Update Proposals
+%%
+\begin{figure}[htb]
+  \begin{align*}
+      & \fun{votedValue_T} \in (\VKeyGen\mapsto\type{T}) \to \type{T}^?\\
+      & \fun{votedValue_T}~\var{vs} =
+        \begin{cases}
+          t & \exists t\in\range{vs}(|vs\restrictrange t|\geq 5) \\
```

Comment:
It seems we miss a `\cdot` between the range of the quantification and its term.